src/sidebar/components/hypothesis-app.js

```diff
@@ -90,16 +90,32 @@ function HypothesisAppController(
     }, 0);
   }
 
-  // Start the login flow. This will present the user with the login dialog.
+  /**
+   * Start the login flow. This will present the user with the login dialog.
+   *
+   * @return {Promise<void>} - A Promise that resolves when the login flow
+   *   completes. For non-OAuth logins, always resolves immediately.
+   */
   this.login = function () {
     if (serviceConfig(settings)) {
```

Comment:
I think that might suggest that the function returns before the login completes, however the returned Promise doesn't resolve until the login completes or fails, even when that requires user interaction in the popup.